com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/util/ComponentsCounter.java

```diff
@@ -323,33 +323,42 @@ public static void copyDataToBuilder(Storage storage, Project project, Collectio
         }
         Map<String, Integer> components = storage.get();
         HashMap<String, Integer> mergedComponents = new HashMap<>();
+        Integer max_instances = 0;
         boolean hasDynamicValue = false;
         for (Map.Entry<String, Integer> entry : components.entrySet()) {
             // different input component names may have the same output name
             // for example wav and sound both are soundc
             String name = project.replaceExt("." + entry.getKey()).substring(1);
             Integer value = entry.getValue();
+            Integer result = 0;
             if (mergedComponents.containsKey(name)) {
                 Integer mergedValue = mergedComponents.get(name);
                 if (mergedValue.equals(DYNAMIC_VALUE) || value.equals(DYNAMIC_VALUE)) {
-                    mergedComponents.put(name, DYNAMIC_VALUE);
+                    result = DYNAMIC_VALUE;
                 } else {
-                    mergedComponents.put(name, mergedValue + value);
+                    result = mergedValue + value;
                 }
             } else {
-                mergedComponents.put(name, value);
+                result = value;
+            }
+            mergedComponents.put(name, result);
+            max_instances = Math.max(max_instances, result);
+            if (isFactoryType(name, true) || hasBones(name)) {
+                hasDynamicValue = true;
             }
-            hasDynamicValue |= isFactoryType(name, true);
-            hasDynamicValue |= hasBones(name);
-        }
-        if (hasDynamicValue) {
-            mergedComponents.put("goc", DYNAMIC_VALUE);
         }
         for (Map.Entry<String, Integer> entry : mergedComponents.entrySet()) {
             ComponenTypeDesc.Builder componentTypeDesc = ComponenTypeDesc.newBuilder();
             componentTypeDesc.setNameHash(MurmurHash.hash64(entry.getKey())).setMaxCount(entry.getValue());
             builder.addComponentTypes(componentTypeDesc);
         }
+        if (hasDynamicValue) {
+            builder.setMaxInstances(DYNAMIC_VALUE);
+            mergedComponents.put("goc", DYNAMIC_VALUE);
+        }
+        else {
+            builder.setMaxInstances(max_instances);
+        }
```

Comment:
`max_instances` - aren't gameobjects. It's better to split these concepts.

e.g. 
-go
-#script
-#script

has one `go` but two `max_instances`. `max_instances` always more or equal to max component. The only place where it's used another way is: 
https://github.com/defold/defold/blob/60c94253d1c2dc427eb34595f68c8017070fd605/engine/gameobject/src/gameobject/comp_anim.cpp#L91-L95

not sure how to seolve it here.